src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java

```diff
@@ -923,6 +919,133 @@ public void siCliccaSullaSecondaPaginaDelleNotifiche(int pagina) {
+        File notificaFile = new File("src/test/resources/notifichePdf/notifica.pdf");
+        String pathNotificaFile = notificaFile.getAbsolutePath();
+        allegatiPASection.caricareNotificaPdfDalComputer(pathNotificaFile);
+
+        if (allegatiPASection.verificaCaricamentoNotificaPdf()) {
+            logger.info("File notifica.pdf caricato correttamente");
+        } else {
+            logger.error("File notifica.pdf non caricato");
+            Assert.fail("File notifica.pdf non caricato");
+        }
+        allegatiPASection.inserimentoNomeAllegato(datiNotificaMap.get("descrizione"));
+    }
+
+    /**
+     * This feature step is used to check the status of the notification after it has been sent through UI.
+     * TODO: This step should be refactored by splitting it into separate methods.
+     */
+    @And("Si verifica che la notifica è stata creata correttamente")
+    public void siVerificaCheLaNotificaeStataCreataCorrettamente() {
+        logger.info("Si verifica che la notifica sia stata creata correttamente filtrandolo per il numero di protocollo");
+        String notificationRequestId = "";
+        for (NetWorkInfo netWorkInfo : netWorkInfos) {
+            if (netWorkInfo.getRequestUrl().contains("/delivery/v2.3/requests") && netWorkInfo.getRequestMethod().equals("POST")) {
+                if (netWorkInfo.getResponseStatus().equals("202") && !netWorkInfo.getResponseBody().isEmpty()) {
+                    notificationRequestId = netWorkInfo.getResponseBody().split("\"notificationRequestId\":\"")[1].split("\"")[0];
+                    logger.info("NotificationRequestId: " + notificationRequestId);
+                    break;
+                }
+            }
+        }
+        if (!notificationRequestId.isEmpty()) {
+            String statusNotifica;
+            int maximumRetry = 0;
+            do {
+                if (maximumRetry > 4) {
+                    logger.error("Sono stati fatti 5 tentativi per verificare la creazione della notifica");
+                    Assert.fail("La notifica risulta ancora in stato WAITING dopo 5 tentativi");
+                }
+                RestNotification restNotification = new RestNotification();
+                statusNotifica = restNotification.getNotificationStatus(notificationRequestId);
+                WebTool.waitTime(90);
+                logger.info("Tentativo n. " + maximumRetry + " - Stato notifica: " + statusNotifica);
+                maximumRetry++;
+            } while (statusNotifica.equals("WAITING"));
+            driver.navigate().refresh();
+            logger.info("La notifica è stata creata correttamente");
+        } else {
+            logger.error("NotificationRequestId non trovato, il codice della risposta al url /delivery/v2.3/requests è diverso di 202 ");
+            Assert.fail("NotificationRequestId non trovato, il codice della risposta al url /delivery/v2.3/requests è diverso di 202 ");
+        }
+    }
```

Comment:
you should write logic step in dedicated page and not in test file, you told me

Reply:
That is true, in fact i add a comment to the step:
https://github.com/pagopa/pn-frontend-e2e/blob/c95c0d8ae40e3d3506c7371c0e01e01e6f709724/src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/NotificaMittentePagoPATest.java#L995-L998
I'll try to find a better solution, right now the `networkInfo` cant be imported outside the `/test` folder